contracts/token/ERC20Permit.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2024 OpenZeppelin, Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {UnstructuredStorage} from "./UnstructuredStorage.sol";
+import {EIP712} from "@openzeppelin/contracts-v4.9/utils/cryptography/EIP712.sol";
+import {IERC2612} from "@openzeppelin/contracts-v4.9/interfaces/IERC2612.sol";
+import {SignatureChecker} from "../lib/SignatureChecker.sol";
+
+contract ERC20Permit is IERC2612, EIP712 {
```

Comment:
might implement https://eips.ethereum.org/EIPS/eip-5267 to make life easier for frontends